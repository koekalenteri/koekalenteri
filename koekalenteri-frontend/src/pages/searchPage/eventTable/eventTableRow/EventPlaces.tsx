import { useMemo } from 'react'
import { useTranslation } from 'react-i18next'
import Box from '@mui/material/Box'
import { Event } from 'koekalenteri-shared/model'

export const EventPlaces = ({ event }: { event: Event }) => {
  const { t } = useTranslation()

  const text = useMemo(() => {
    if (event.places) {
      if (event.entries) {
        return `${event.entries} / ${event.places}`
      }
      return event.places + ' ' + t('toltaPlaces')
    }
    return ''
  }, [event.entries, event.places, t])

  return <Box textAlign="right">{text}</Box>
}
