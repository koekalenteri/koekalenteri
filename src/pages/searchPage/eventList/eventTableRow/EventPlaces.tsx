import type { PublicDogEvent } from '../../../../types'

import { useMemo } from 'react'
import { useTranslation } from 'react-i18next'
import Box from '@mui/material/Box'

import { isEntryOpen } from '../../../../lib/utils'

export const EventPlaces = ({ event }: { event: PublicDogEvent }) => {
  const { t } = useTranslation()

  const text = useMemo(() => {
    if (event.places) {
      if (event.entries || isEntryOpen(event)) {
        return `${event.entries ?? 0} / ${event.places}`
      }
      return event.places + ' ' + t('toltaPlaces')
    }
    return ''
  }, [event, t])

  return <Box textAlign="right">{text}</Box>
}
