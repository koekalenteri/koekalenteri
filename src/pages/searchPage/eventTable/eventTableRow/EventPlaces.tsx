import type { PublicDogEvent } from '../../../../types'

import { useMemo } from 'react'
import { useTranslation } from 'react-i18next'
import Box from '@mui/material/Box'

export const EventPlaces = ({ event }: { event: PublicDogEvent }) => {
  const { t } = useTranslation()

  const text = useMemo(() => {
    if (event.places) {
      if (event.entries) {
        return `${event.entries} / ${event.places}`
      }
      return event.places + ' ' + t('toltaPlaces')
    }
    return ''
  }, [event.entries, event.places, t])

  return <Box textAlign="right">{text}</Box>
}
