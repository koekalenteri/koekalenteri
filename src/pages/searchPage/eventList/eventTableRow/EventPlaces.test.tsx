import { render, screen } from '@testing-library/react'

import {
  eventWithEntryClosed,
  eventWithEntryOpen,
  eventWithEntryOpenButNoEntries,
} from '../../../../__mockData__/events'

import { EventPlaces } from './EventPlaces'

describe('EventInfo', () => {
  it('render places for a class in event with multiple classes', async () => {
    const { container } = render(<EventPlaces event={eventWithEntryClosed} />)

    expect(container).toMatchSnapshot()
    expect(screen.getByText('4 / 4')).toBeInTheDocument()
  })

  it('render places for event with single class and places defined only on event level', async () => {
    const { container } = render(<EventPlaces event={eventWithEntryOpen} />)

    expect(container).toMatchSnapshot()
    expect(screen.getByText('7 / 10')).toBeInTheDocument()
  })

  it('render places for event entry open but no entries', async () => {
    const { container } = render(<EventPlaces event={eventWithEntryOpenButNoEntries} />)

    expect(container).toMatchSnapshot()
    expect(screen.getByText('0 / 10')).toBeInTheDocument()
  })
})
