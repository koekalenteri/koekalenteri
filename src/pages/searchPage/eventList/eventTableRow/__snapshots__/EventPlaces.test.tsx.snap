// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventInfo render places for a class in event with multiple classes 1`] = `
<div>
  <div
    class="MuiBox-root css-s2uf1z"
  >
    4 / 4
  </div>
</div>
`;

exports[`EventInfo render places for event entry open but no entries 1`] = `
<div>
  <div
    class="MuiBox-root css-s2uf1z"
  >
    0 / 10
  </div>
</div>
`;

exports[`EventInfo render places for event with single class and places defined only on event level 1`] = `
<div>
  <div
    class="MuiBox-root css-s2uf1z"
  >
    7 / 10
  </div>
</div>
`;
